### DISCUSSION TITLE
Tomb Raider benchmark

### DISCUSSION THREAD

**Discussion by WSLUser** [microsoft/wslg#784] [Ideas]
Hear me out here,  I know it's a game and older too but still plenty resource heavy at ultra settings today and was first game to use TressFX which is utilized by the benchmark.  I think it's a pretty good one to do reasonably well on. If you can manage to get 30 FPS at max settings then you're pretty much gold for most apps. Not that certain tunings or optimizations won't be needed but majority of GUI software ought to run just fine without major perf penalties. I could even mention graphic protocols like Sixel used by terminals in WSL2 that would hugely benefit from getting this benchmark up to FPS rate I mentioned. Granted one of the biggest hurdles would be anti-aliasing. You'd need to do well with MSAA, FXAA and other often proprietary formats such as TXAA (Nvidia) to start with. So any takers for seeing how horribly slow it is currently (if it doesn't crash that is). Also require use of Wine, I'd use Valve's fork which has optimizations meant for games though provides better app support in general.

**Comment by crramirez** [comment]
Hello @WSLUser,

One question, did you make the benchmark or suggest it? It is not clear to me.

**Reply by WSLUser** [reply]
Suggesting it. Have you ever played Tomb Raider (from 2014) before? As the first game to feature AMD's brand new release of TressFX, which was created to replace the physics that until then was being provided only by Nvidia's PhysX engine, it provided more realistic hair and possibly grass as well (I forget when they added grass support). Because of this, the game developers added an in-game benchmark you could run from the main menu prior to actually starting the game and provided an FPS count which you could also visualize as it moved through different scenes. I think there was a mini FPS chart as well that updated results in real-time. You could test with both V-Sync on and off by simply switching that setting from the Settings menu in the main menu. V-sync is actually a desired thing to have but some games required it to be off to get the full performance depending on various factors including hardware. These days it comes enabled by default unless you have something globally disabling like with Nvidia's Control Panel. 

There are actually quite a few games which have provided in-game benchmarks and I wouldn't be surprised if a different game was chosen if they choose to run one at all. But I think the Tomb Raider one is best as it's the only I know of which allows you to test with both PhysX and TressFX. Keep in mind that as TressFx was much more limited then, whatever wasn't being provided by it, was getting provided by PhysX. It was considered more of a novelty then but there were very noticeable differences that made the game appear much better for what it could do.

**Reply by WSLUser** [reply]
So you can get a better idea about TressFX and what it does. Nice to see the improvements they've made. I haven't checked on it in awhile. I just wish news outlets were more informative about the games using this engine vs PhysX. https://gpuopen.com/tressfx/ https://github.com/GPUOpen-Effects/TressFX
